cmd/memparse.go

```diff
@@ -273,3 +294,40 @@ func generateHexAndAscii(data []byte) (string, string) {
 
 	return hex, ascii
 }
+
+func printMemorySearchResult(task internal.Task) error {
+	memReader, err := crit.NewMemoryReader(
+		filepath.Join(task.OutputDir, metadata.CheckpointDirectory),
+		*pID, pageSize,
+	)
+	if err != nil {
+		return err
+	}
+
+	if err := internal.UntarFiles(
+		task.CheckpointFilePath, task.OutputDir,
+		[]string{filepath.Join(metadata.CheckpointDirectory, fmt.Sprintf("pages-%d.img", memReader.GetPagesID()))},
+	); err != nil {
+		return err
+	}
+
+	matches, err := memReader.SearchPattern(*searchPattern, *searchContext)
+	if err != nil {
+		return err
+	}
+
+	if len(matches) == 0 {
+		fmt.Printf("\nNothing found for pattern %s in memory pages of process %d", *searchPattern, *pID)
+		return nil
+	}
+
+	patternLength := len(*searchPattern) + (*searchContext * 2)
+
+	fmt.Printf("\n%-16s  %-*s  %-5s\n", "Address", patternLength, "Pattern", "Instance")
```

Comment:
`Pattern` is slightly misleading here. That would be the string supplied by the user - I feel this would be better named as `Result` or `Match`.